### ISSUE TITLE
Suport docker --cidfile option

### ISSUE THREAD

**Issue by Grokzen** [docker/compose#940]
`docker run` has the flag `--cidfile <file>` that writes the container id to the specefied file so that it is possible to read it for other purposes.

It should be a option in the config for one image

``` yaml
web:
  image: mysql:latest
  cidfile: /tmp/mysql.cid
...
```

Supporting this might be a simpler solution to build before `exec` command is supported (as discussed in https://github.com/docker/fig/issues/593)

**Comment by mavenugo** [comment]
:+1:

**Comment by tobowers** [comment]
:+1:

**Comment by stonemaster** [comment]
+1

**Comment by jamessmoore** [comment]
I encountered this as well +1

**Comment by MateusAmin** [comment]
+1

**Comment by OferE** [comment]
+1

**Comment by kaja-mohideen** [comment]
+1

**Comment by joshuacox** [comment]
this one is a funny one, I see @MateusAmin's point about services.  And certainly what happens when you scale a service?  Do subsequent cidfiles get a -xxx number added to them? Or does the collection of services itself get a collective pid? Should all these pidfiles be kept in a subdirectory?  Possibly .hidden?    I certainly appreciate having the option to create one of these things as much of the early life of my images uses a Makefile for testing which saves me a ton of typing e.g.

```
cid:
    docker run --cidfile=cid -it mything

logs:
    docker logs `cat cid`

enter:
    docker exec -it `cat cid` /bin/bash
```

now I just:
```
make cid
make logs
make enter
```

to do the things I do repeatedly while creating, developing and testing containers

I'll put a +.5 as I want to watch this issue.

**Comment by AndreLouisCaron** [comment]
@joshuacox The CID file contains the container ID.  When you scale a service, you get multiple containers and each has its unique ID, right?  Why would there be any kind of `-xxx` suffix?

**Comment by joshuacox** [comment]
@AndreLouisCaron  I wish it worked that way, as far as I know docker-compose still does not support a `-cidfile` option?

 To this day I still use my Makefile method when dev'ing locally on many of projects in their early stages.  Or in the case of [Chromate](https://github.com/joshuacox/Chromate) container, which is something I use locally to try out running chrome in a container on my desktop, and I'll never run it on a `server`, I'll likely not use anything else.  Mainly because I use the cidfile for the `make logs` and `make enter` methods I detailed above.



EDIT:  I see your commit now to compose, I eagerly await this to be merged into production, bravo!

**Comment by AndreLouisCaron** [comment]
@joshuacox There doesn't seem to be much interest to get that merged and it doesn't do what I want in the end.  However, if someone is interested in pushing the maintainers to accept the patch, then I'll gladly see it through to completion.

**Comment by joshuacox** [comment]
@AndreLouisCaron  I have interest for it, and honestly this is one of the reasons why I have never adopted docker-compose (and fig before it) largely (unless it is managed by rancher that is, and I never use docker-compose).  Ever since the beginning I've seen docker-compose fail randomly as it does not properly wait on dependency containers (for example mysql as a container, works great on idle machines, but if a docker-compose is used on a heavily laden machine the mysql will not start quick enough and the docker-compose has no clue how to wait for it to properly start).

**Comment by rdxmb** [comment]
:+1:

**Comment by Persik3D** [comment]
👍
